### PR TITLE
[REVIEW] Remove cxx11 abi handling from CMake

### PR DESCRIPTION
**Pull request by kkraus14**
We historically had a CMake variable to control the cxx11 ABI as we used to build against old ABI projects in Ubuntu, which by default uses the new ABI. This removes this option from the CMake as its no longer used.

If someone absolutely needs the old ABI they can still manually pass in compilers flags via variables like `CMAKE_C_FLAGS`, `CMAKE_CXX_FLAGS`, `CMAKE_CUDA_FLAGS`, etc.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by kkraus14** [comment]
CI failure is from a flaky jit multiprocess test.

**Review by jlowe — APPROVED** [review]

**Review by jlowe — CHANGES_REQUESTED** [review]
I just noticed `CONTRIBUTING.md` references the `CMAKE_CXX11_ABI` flag that is being removed, that should be updated as well.

**Comment by kkraus14** [comment]
> I just noticed `CONTRIBUTING.md` references the `CMAKE_CXX11_ABI` flag that is being removed, that should be updated as well.

Fixed as well as handling some `libcudf_kafka` stuff I missed as well.

**Review by jlowe — APPROVED** [review]

**Comment by kkraus14** [comment]
cc @harrism to review

File: conda/recipes/libcudf/meta.yaml [hunk 1]
```diff
@@ -38,7 +38,7 @@ requirements:
     - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}
     - dlpack
     - arrow-cpp-proc * cuda
-    - boost-cpp >=1.72.0
+    - {{ pin_compatible('boost-cpp', max_pin='x.x') }}
```
Comment (jakirkham) [inline conda/recipes/libcudf/meta.yaml]: Would suggest pinning to the patch release as they include the patch release in the SONAME as well.

```suggestion
    - {{ pin_compatible('boost-cpp', max_pin='x.x.x') }}
```

https://abi-laboratory.pro/index.php?view=timeline&l=boost

Comment (jakirkham) [reply conda/recipes/libcudf/meta.yaml]: Lol fixed after clicking submit. Marking as resolved

**Review by jakirkham — APPROVED** [review]

File: conda/recipes/libcudf/meta.yaml [hunk 2]
```diff
@@ -38,7 +38,7 @@ requirements:
     - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}
     - dlpack
     - arrow-cpp-proc * cuda
-    - boost-cpp >=1.72.0
+    - {{ pin_compatible('boost-cpp', max_pin='x.x.x') }}
```
Comment (jakirkham) [inline conda/recipes/libcudf/meta.yaml:41]: Note this is needed as `boost-cpp` doesn't currently have `run_exports` ( https://github.com/conda-forge/boost-cpp-feedstock/pull/82 ).

**Review by raydouglass — APPROVED** [review]

**Review by jrhemstad — APPROVED** [review]

**Review by harrism — APPROVED** [review]

**Pull request merged**